sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -1703,6 +1704,26 @@ public void populateDisplayData(DisplayData.Builder builder) {
         .build();
   }
 
+  /**
+   * A {@link Sink} for use with {@link FileIO#write} and {@link FileIO#writeDynamic}, writing
+   * elements with a given (common) schema, like {@link #writeGenericRecords(Schema)}.
+   */
+  public static <ElementT extends IndexedRecord> Sink<ElementT> sinkViaGeneric(Schema schema) {
```

Comment:
FYI: I picked `sinkViaGeneric` deliberately instead of GenericRecord -- I'm taking a look whether this sink can be used with any generic data that matches the schema (although avro records are the most frequently seen...)

Reply:
maybe let's call it `sinkWithSchema`

Reply:
Mark it too as `@Experimental` please.